### PR TITLE
[STG-1450] Fix Session init failing when no pages are open, falls back to opening about:blank

### PR DESCRIPTION
**Pull request by pirate**
# why

Stagehand API failures were caused on all routes `/act`, `/observe`, etc. when session init failed because a page was closed. This adds a fallback to open about:blank so at least the session remains usable instead of just crashing on init.

# what changed

# test plan

---
## Summary by cubic
Fixes session init when connecting to a browser with zero pages by auto-creating an about:blank tab. Keeps Stagehand routes working after the last page closes and surfaces lint failures via a dedicated cancellation job. Addresses STG-1450.

- **Bug Fixes**
  - Replace waitForFirstTopLevelPage with ensureFirstTopLevelPage: check for an existing page, wait with timeout, then create about:blank if none.
  - Simplify errors to a TimeoutError with clearer context; improve logging and add an integration test for zero-page init.

- **CI**
  - Run cancellation in a separate job only when lint fails, so failures are visible instead of marked skipped.

<sup>Written for commit 45ed6dc059d44f07c338dc8f36e72eb0702a2cfd. Summary will update on new commits. <a href="https://cubic.dev/pr/browserbase/stagehand/pull/1749">Review in cubic</a></sup>

Mintlify
---
0 threads from 0 users in Mintlify

- No unresolved comments

<a href="https://dashboard.mintlify.com/browserbase/stagehand/editor/session-init-fixes?source=pr_comment" target="_blank" rel="noopener noreferrer"><picture><source media="(prefers-color-scheme: dark)" srcset="https://d3gk2c5xim1je2.cloudfront.net/assets/open-mintlify-editor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://d3gk2c5xim1je2.cloudfront.net/assets/open-mintlify-editor-light.svg">Open in Mintlify Editor</picture></a>

### CODE REVIEW

File: packages/core/lib/v3/understudy/context.ts [hunk 1]
```diff
@@ -75,6 +75,14 @@ const DEFAULT_FIRST_TOP_LEVEL_PAGE_TIMEOUT_MS = 5000;
 const CI_FIRST_TOP_LEVEL_PAGE_TIMEOUT_MS = 30000;
 const FIRST_TOP_LEVEL_PAGE_TIMEOUT_ENV =
   "STAGEHAND_FIRST_TOP_LEVEL_PAGE_TIMEOUT_MS";
+const WAIT_FOR_FIRST_TOP_LEVEL_PAGE_OPERATION =
+  "waitForFirstTopLevelPage (no top-level Page)";
+
+class WaitForFirstTopLevelPageTimeoutError extends TimeoutError {
```
Comment (seanmcguire12) [inline packages/core/lib/v3/understudy/context.ts]: i dont think we need another error class for this unless you feel strongly. either way, it should go inside the `packages/core/types/sdkErrors.ts` & be exported with the other error types

Comment (pirate) [reply packages/core/lib/v3/understudy/context.ts]: it's internal because users will never see it, it's a class because we dont want to do string comparison to catch this specific error
